scheduler/scheduler.go

```diff
@@ -346,3 +360,23 @@ func maxDepth(tables schema.Tables) uint64 {
 	}
 	return depth
 }
+
+func shardTableClients(tableClients []tableClient, shard *shard) []tableClient {
+	if shard == nil || len(tableClients) == 0 {
+		return tableClients
+	}
+	num := int(shard.num)
+	total := int(shard.total)
+	chunkSize := len(tableClients) / total
+	if chunkSize == 0 {
+		chunkSize = 1
+	}
+	chunks := lo.Chunk(tableClients, chunkSize)
+	if num > len(chunks) {
+		return nil
```

Comment:
Shouldn't this be an error?

Reply:
I think an empty shard (e.g. 10 table clients, 100 shards) shouldn't error out, it simply shouldn't do anything